### ISSUE TITLE
Truncated fields

### ISSUE THREAD

**Issue by amphibithen** [sivasamyk/logtrail#209]
I am using Kibana 5.6.3, Elasticsearch 5.6.3, logtrail 0.1.23, and FluentD 0.12.  I am monitoring docker container logs, and have data such as docker image name, and docker container id. my logs look something like this::

![logtrail_truncate](https://user-images.githubusercontent.com/9948555/33332908-7baaf89a-d433-11e7-8fb1-4707465c2242.png)

My logtrail.json has this::
```
{
  "version" : 1,
  "index_patterns" : [
    {      
      "es": {
        "default_index": "logstash-*"
      },
      "tail_interval_in_seconds": 10,
      "es_index_time_offset_in_seconds": 0,
      "display_timezone": "local",
      "display_timestamp_format": "MMM DD HH:mm:ss",
      "max_buckets": 500,
      "default_time_range_in_days" : 0,
      "max_hosts": 100,
      "max_events_to_keep_in_viewer": 5000,
      "fields" : {
        "mapping" : {
            "timestamp" : "@timestamp",
            "display_timestamp" : "@timestamp",
            "hostname" : "container_image",
            "program": "container_id",
            "message": "log"
        },
        "message_format": "{{{log}}}"
      },
      "color_mapping" : {
      }
    }  
  ]
}
```

What i am trying to do is truncate the container_image and container_id.  i tried creating kibana scripted field, but it appears logtrail cannot pick that up.  how to display truncated values?

**Comment by sivasamyk** [comment]
Logtrail does not pickup data from .kibana index. May be you can add extra field with truncated value in ES using FluentD (I have not used fluentd. Not sure if it is possible)

**Comment by philomory** [comment]
I'd love if there were a way to handle this within logtrail itself, as this is primarily a display issue. That is, if I use some regex trickery in (for example) logstash to get the last 6 characters of a long identifier, I can stick that in a field and use it in logtrail's `message_format` option, but then when I click the resulting link in logtrail it's going to filter by only the last 6 characters, whereas I most likely want to filter on the whole identifier.

Ideally the message_format string would be expanded to accept something like `{{{pid|short-pid}}}`, which would result in `<a class="ng-binding" ng-click="onClick('pid','{{pid}}')">{{short-pid}}</a>`. A normal `{{{pid}}}` would work exactly like it does now.

This logic of doing it this way (rather than going for something like `{{{pid|maxlength:6}}}` or whatever) is that it allows the admin to control the display of links independently from their target, without forcing logtrail itself to implement any specific formatting functions like `truncate` (which would just lead to an ever-increasing list of requests for `upcase`, `downcase`, `reverse`, and so on).

**Comment by amphibithen** [comment]
i agree that adding extra fields to the index seems excessive, since its a display issue.  you cannot tell from the screenshot i provided, but on average half of the screen is taken up by the `time + docker image + container ID`, which means the log lines only start to show up before they get wrapped and looks very ugly :)

i like your idea philomory, but at present i think the only solution is to implement what's suggested by sivasamyk.

can we turn this into a feature request?
